WordPress/src/test/java/org/wordpress/android/ui/stories/SaveStoryGutenbergBlockUseCaseTest.kt

```diff
@@ -0,0 +1,288 @@
+        )
+
+        // Then
+        Assertions.assertThat(result).isEqualTo(blockWithEmptyMediaFiles)
+    }
+
+    @Test
+    fun `obtain a mediaFileData from a MediaFile with a temporary id`() {
+        // Given
+        val mediaFile = getOneMediaFile(1)
+
+        // When
+        val mediaFileData = saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryId(
+                mediaFile,
+                TEMPORARY_ID_PREFIX + 1
+        )
+
+        // Then
+        Assertions.assertThat(mediaFileData.alt).isEqualTo("")
+        Assertions.assertThat(mediaFileData.id).isEqualTo(TEMPORARY_ID_PREFIX + 1)
+        Assertions.assertThat(mediaFileData.link).isEqualTo(
+                "https://testsite.files.wordpress.com/2020/10/wp-0000000.jpg"
+        )
+        Assertions.assertThat(mediaFileData.type).isEqualTo("image")
+        Assertions.assertThat(mediaFileData.mime).isEqualTo(mediaFile.mimeType)
+        Assertions.assertThat(mediaFileData.caption).isEqualTo("")
+        Assertions.assertThat(mediaFileData.url).isEqualTo(mediaFile.fileURL)
+    }
+
+    @Test
+    fun `local media id is found and gets replaced with remote media id`() {
+        // Given
+        val mediaFile = getOneMediaFile(1)
+        val onePost = PostModel()
+        onePost.setContent(blockWithNonEmptyMediaFiles)
+
+        // When
+        saveStoryGutenbergBlockUseCase.replaceLocalMediaIdsWithRemoteMediaIdsInPost(
+                onePost,
+                mediaFile
+        )
+
+        // Then
+        Assertions.assertThat(onePost.content).isEqualTo(blockWithNonEmptyMediaFilesWithOneRemoteId)
+    }
+
+    @Test
+    fun `slides are saved locally to storiedPrefs`() {
+        // Given
+        val frames = ArrayList<StoryFrameItem>()
+        frames.add(getOneStoryFrameItem("1"))
+        frames.add(getOneStoryFrameItem("2"))
+        frames.add(getOneStoryFrameItem("3"))
+
+        // When
+        saveStoryGutenbergBlockUseCase.saveNewLocalFilesToStoriesPrefsTempSlides(
+                siteModel,
+                0,
+                frames
+        )
+
+        // Then
+        verify(storiesPrefs, times(3)).saveSlideWithTempId(any(), any(), any())
+    }
+
+    private fun setupFluxCMediaFiles(
+        emptyList: Boolean
+    ): ArrayList<MediaFile> {
+        when (emptyList) {
+            true -> return ArrayList()
+            false -> {
+                val mediaFiles = ArrayList<MediaFile>()
+                for (i in 1..10) {
+                    val oneMediaFile = MediaFile()
+                    oneMediaFile.id = i
+                    oneMediaFile.mediaId = (i + 1000).toString()
+                    oneMediaFile.mimeType = "image/jpeg"
+                    oneMediaFile.fileURL = "https://testsite.files.wordpress.com/2020/10/wp-0000000.jpg"
+                    mediaFiles.add(oneMediaFile)
+                }
+                return mediaFiles
+            }
+        }
+    }
+
+    private fun getOneMediaFile(id: Int): MediaFile {
```

Comment:
We can put a default value of 1 here for this `id` since it's always 1 in this test.

Reply:
I prefer to keep it explicit if we needed to extend it, since for example there's a correlation between the id given in the constructor param and the way things are setup in [this line here](https://github.com/wordpress-mobile/WordPress-Android/pull/13103/files#diff-55f5e21ccf6cc6baa9e15eb33285d543008f79657547a539be6b68ef329d3283R115) where the `+ 1`   in the `TEMPORARY_ID_PREFIX + 1` is relevant and needs to match the value passed in the constructor. Sounds good?

Reply:
oh I get you. In that case let's put the 1 in a variable and the temp in one as well for reusability and clarity. 

so something like 
```
        // Given
        val mediaFileId = 1
        val mediaFile = getOneMediaFile(mediaFileId)
       val tempMediaFileId = TEMPORARY_ID_PREFIX + mediaFileId 
```

Please let me know if this is helpful. Thank you.

Reply:
> Please let me know if this is helpful. 

It is! done in 621abbf388